### PR TITLE
tests: ts2: adapt fuzzer

### LINKED ISSUE #7342: TS2: adapt fuzzer

### PR DESCRIPTION
**Pull request by eckter**
1. Add a new version of the fuzzer script, calling only timetable v2 versions of the endpoints
2. Add a new version of the script loading the fuzzer files to reproduce the test cases

The old version should be deleted at the same time as the old endpoints. 

One difference compared to the v1 is that we don't ignore any error, we don't ignore when the path doesn't exist or when the margins are impossible. It's not necessarily a problem (we don't risk ignoring an actual issue), but we need to filter through the results manually. And we can't run the script directly in the test suite anymore.

### CODE REVIEW

File: tests/fuzzer/fuzzer_v2.py [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+            raise RuntimeError(f"Infra error {r.status_code}: {r.content}")
+        rjson = r.json()
+        for infra in rjson["results"]:
+            if infra["name"] == infra_name:
+                return infra["id"]
+        page = rjson.get("next")
+    raise ValueError(f"Unable to find infra {infra_name}")
+
+
+def create_scenario(editoast_url: str, infra_id: int) -> Scenario:
```
Comment (flomonster) [inline tests/fuzzer/fuzzer_v2.py:112]: We don't need to have a Project / Study / Scenario anymore. We can have a timetable alone.

File: tests/fuzzer/fuzzer.py [hunk 2]
```diff
@@ -282,7 +283,7 @@ def run(
                 print(e)
                 log_folder.mkdir(exist_ok=True)
                 with open(str(log_folder / f"{i}.json"), "w") as f:
-                    print(json.dumps(e.args[0], indent=4), file=f)
+                    print(json.dumps(e.args[0], indent=4, default=lambda o: "<not serializable>"), file=f)
```
Comment (shenriotpro) [inline tests/fuzzer/fuzzer.py:288]: nit: would you have an example of something we can't serialize?

File: tests/tests/test_regressions.py [hunk 3]
```diff
@@ -14,6 +14,7 @@
 REGRESSION_TESTS_JSON_FILES = [
     str(json_file.relative_to(REGRESSION_TESTS_DATA_FOLDER))
     for json_file in REGRESSION_TESTS_DATA_FOLDER.rglob("*.json")
+    if json.loads(json_file.read_bytes()).get("timetable_version") != 2
```
Comment (shenriotpro) [inline tests/tests/test_regressions.py]: nit: do we need read_bytes? Would something like read work?

File: tests/tests/test_regressions_v2.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    Send a request to preload the infra, raise a RuntimeError on failure
+    """
+    r = requests.post(editoast_url + f"infra/{infra_id}/load")
+    if r.status_code // 100 != 2:
+        if r.status_code // 100 == 4:
+            return None
+        raise RuntimeError(f"Infra load {r.status_code}: {r.content}")
+
+
+def _schedule_with_payload(editoast_url: str, payload, accept_400, scenario: Scenario):
```
Comment (shenriotpro) [inline tests/tests/test_regressions_v2.py]: can we add a type annotation for the output?

Comment (shenriotpro) [reply tests/tests/test_regressions_v2.py]: (same for other args)

File: tests/tests/test_regressions.py
Comment (shenriotpro) [inline tests/tests/test_regressions.py:1]: can we add a note explaining this is deprecated and should be removed when the migration is done?

File: tests/fuzzer/fuzzer_v2.py [hunk 5]
```diff
@@ -0,0 +1,629 @@
...
+        self.links[b].append(a)
+
+
+@dataclass
+class _RollingStock:
+    name: str
+    id: int
+
+
+def run(
```
Comment (shenriotpro) [inline tests/fuzzer/fuzzer_v2.py:39]: nit: can we move these "public" functions above the "private" classes?

File: tests/fuzzer/fuzzer_v2.py [hunk 6]
```diff
@@ -0,0 +1,629 @@
...
+        switch_type = builtin_node_types()[switch["switch_type"]]
+        for group in switch_type["groups"].values():
+            for pair in group:
+                src = _TrackEndpoint.from_dict(switch["ports"][pair["src"]])
+                dst = _TrackEndpoint.from_dict(switch["ports"][pair["dst"]])
+                graph.link(src, dst)
+    return graph
+
+
+def _random_set_element(s: Iterable):
```
Comment (shenriotpro) [inline tests/fuzzer/fuzzer_v2.py]: nit: perhaps we could annotate the output thanks to TypeVar https://docs.python.org/3/library/typing.html#generics

File: tests/fuzzer/fuzzer.py (hunk 2 above)
Comment (eckter) [reply tests/fuzzer/fuzzer.py:288]: Some uncommon exceptions can't be serialized trivially (like broken pipes and such), having a default here avoid having the whole script crashing

**Review by shenriotpro — APPROVED** [review]

**Pull request merged**